src/histolab/tiler.py

```diff
@@ -100,19 +122,62 @@ def locate_tiles(
         img.putalpha(alpha)
         draw = PIL.ImageDraw.Draw(img)
 
-        tiles = (
-            self._tiles_generator(slide, extraction_mask)[0]
-            if isinstance(self, ScoreTiler)
-            else self._tiles_generator(slide, extraction_mask)
-        )
+        if tiles is None:
+            tiles = (
+                self._tiles_generator(slide, extraction_mask)[0]
+                if isinstance(self, ScoreTiler)
+                else self._tiles_generator(slide, extraction_mask)
+            )
         tiles_coords = (tile[1] for tile in tiles)
-        for coords in tiles_coords:
+
+        for coords, one_outline in self._tile_coords_and_outline_generator(
```

Comment:
I'm not sure `_tile_coords_and_outline_generator` is the right approach here. My general concern is that basically we are checking for the outline type and then we return the coord and the outline as they are passed to the method if they respect all the constraint. So it's just a validator, IMHO i do prefer to have a validator as you had in one of your pervious commit. Also i think that we could have loss of performance with a large number of Tiles and outlines 🤷🏽‍♂️

Reply:
I beg to differ here. It's not just a validator .. depending on the type of ``outline`` argument, it either calls ``for coords in tiles_coords`` or ``for coords, one_outline in zip_longest(tiles_coords, outlines)``. So if the outline is a ``str`` it would return the same value for all coords, whereas if it is an ``Iterable`` it would fetch the value corresponding to the relevant tile and throw an exception is the lengths don't match. Even if I wanted to use a validator I couldn't because the number of tiles is not known beforehand in ``GridTiler`` so the only way to check the difference in length between it and the ``outline`` ``Iterable`` is using something like the ``zip_longest``.

Reply:
__gotcha__ ok